apps/src/templates/VersionHistory.jsx

```diff
@@ -62,10 +66,31 @@ module.exports = React.createClass({
     this.setState({versions: null});
   },
 
+  onConfirmClearPuzzle: function () {
+    this.setState({confirmingClearPuzzle: true});
+  },
+
+  onCancelClearPuzzle: function () {
+    this.setState({confirmingClearPuzzle: false});
+  },
+
+  onClearPuzzle: function () {
+    this.props.handleClearPuzzle();
+    this.props.handleCloseDialog();
+  },
+
   render: function () {
     var versionList;
+    if (this.state.confirmingClearPuzzle) {
+      versionList = (
+        <div>
+          <p>Are you sure you want to clear all Big Game progress&#63;</p>
+          <button id="confirm-button" style={{float: 'right'}} onClick={this.onClearPuzzle}>Start Over</button>
```

Comment:
If version history is going to be used eventually for all projects, I think these buttons need to be localized. I don't think that stops this from getting merged, since right now it's only CSinA and CSP and is already out in English. I'll just make a note for later.